applications/FluidDynamicsApplication/custom_utilities/mass_conservation_utility.cpp

```diff
@@ -246,43 +302,112 @@ void MassConservationCheckProcess::ComputeVolumesAndInterface( double& positiveV
             }
-        }
 
-        if ( pt_count_pos == rGeom.PointsNumber() || pt_count_neg == rGeom.PointsNumber() ){
-            // all nodes are on one side and the element is not split
-            continue;
+            if ( r_geom.PointsNumber() == 3 ){ p_modified_sh_func = Kratos::make_unique<Triangle2D3ModifiedShapeFunctions>(it_elem->pGetGeometry(), Distance); }
+            else if ( r_geom.PointsNumber() == 4 ){ p_modified_sh_func = Kratos::make_unique<Tetrahedra3D4ModifiedShapeFunctions>(it_elem->pGetGeometry(), Distance); }
+            else { KRATOS_ERROR << "The utility can not be applied on this kind of element" << std::endl; }
+
+            // Concerning their area, the positive and negative side of the interface are equal
+            p_modified_sh_func->ComputeInterfacePositiveSideShapeFunctionsAndGradientsValues(
+                    shape_functions,                    // N
+                    shape_derivatives,                  // DN
+                    w_gauss_interface,                  // includes the weights of the GAUSS points (!!!)
+                    GeometryData::GI_GAUSS_2);          // second order Gauss integration
+
+            // negative side outwards area normal vector values for the Gauss pts. of given quadrature
+            std::vector<Vector> normal_vectors;
+            p_modified_sh_func->ComputeNegativeSideInterfaceAreaNormals(
+                normal_vectors,
+                GeometryData::GI_GAUSS_2
+            );
+
+            // iteration over all 3 or 6 integration points
+            for ( unsigned int i_gauss = 0; i_gauss < w_gauss_interface.size(); i_gauss++)
+            {
+                const double& r_weight = w_gauss_interface[i_gauss];
+                const auto& r_N = row(shape_functions, i_gauss);
+
+                auto& r_normal = normal_vectors[i_gauss];
+                r_normal /= norm_2( r_normal );
+                r_normal *= Factor;
+
+                array_1d<double,3> interpolated_velocity = ZeroVector(3);
+                for (unsigned int n_node = 0; n_node < r_geom.PointsNumber(); n_node++){
+                    noalias( interpolated_velocity ) += r_N[n_node] * (r_geom[n_node].FastGetSolutionStepValue(VELOCITY)
+                        - r_geom[n_node].FastGetSolutionStepValue(MESH_VELOCITY));
+                }
+
+                const double& r_orthogonal_flow = inner_prod( r_normal, interpolated_velocity );
+                // checking if it is really an outflow towards the air domain
+                if ( 0.0 < r_orthogonal_flow ){
+                    outflow += r_weight * r_orthogonal_flow;
+                }
+            }
         }
-        else {
+    }
+    mrModelPart.GetCommunicator().SynchronizeCurrentDataToMin(DISTANCE);
+    const double global_outflow = mrModelPart.GetCommunicator().GetDataCommunicator().SumAll(outflow);
+    return global_outflow;
+}
+
+
+
+double MassConservationUtility::ComputeInterfaceArea(){
```

Comment:
Maybe we can find a more descriptive name. Something like `CalculateTotalInterfaceArea` (what comes to my mind with this name is a single element area).

Reply:
this function is actually not used anymore, removed